MetaMask/metamask-extension
# chore: smart-transactions code domain

## **Description**

This PR establishes the `ui/pages/smart-transactions` code domain, following the guidance in [this proposal](https://docs.google.com/document/d/1dzgWc1LCkYz2lZZ6X8SOaj23iaGYVQ2zJ1FVpOxsU-o/edit).


Moved | To ui/pages/smart-transactions/
-- | --
ui/components/app/smart-transactions/smart-transactions-opt-in-modal.tsx | components/smart-transactions-opt-in-modal.tsx

The following files will be left in place.
| Left in place                                                                                  | Reason                                                                        |
|---------------------------------------------------------------------------------------|-------------------------------------------------------------------------------|
app/scripts/lib/transaction/smart-transactions.ts | background code
shared/constants/smartTransactions.ts | shared with background
| `ui/components/app/transaction-list-item/smart-transaction-list-item.component.js`     | Specific to the transaction list.                                             |
| `ui/pages/confirmations/confirmation/templates/smart-transaction-status-page.js`      | Belongs with other confirmation templates.                                    |
| `ui/pages/swaps/smart-transaction-status/smart-transaction-status.js`                 | Swaps-focused. Will be replaced by regular STX status screen once swaps moves to regular STX. |

### STX-related code in other files
The following files contain stx-related code, which may be extracted and moved into the smart-transactions code domain at a later time.

| Path                                                                                     | STX-Related Code                                                                                                                                 |
|------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
| `app/scripts/metamask-controller.js`                                                     | We use the STX controller there `new SmartTransactionsController`. `*// Smart Transactions` section*                                             |
| `ui/ducks/swaps/swaps.js`                                                                | A few functions related to STX: `fetchSmartTransactionsLiveness`, `signAndSendSmartTransaction`, `setSmartTransactionsRefreshInterval`, etc.     |
| `ui/store/actions.ts`                                                                    | `fetchSmartTransactionsLiveness` and other STX functions                                                                                         |
| `ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.container.js`  | STX is used from this component                                                                                                                  |
| `ui/pages/settings/advanced-tab/advanced-tab.component.js`                               | Advanced Settings that include STX option                                                                                                        |
| `ui/pages/swaps/swaps.util.ts`                                                           | `getFeeForSmartTransaction`                                                                                                                      |
| `ui/pages/swaps/prepare-swap-page/review-quote.js`                                       | Uses STX in Swaps                                                                                                                                 |
| `ui/selectors/transactions.js`                                                           | `allowedSwapsSmartTransactionStatusesForActivityList`                                                                                            |




[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26948?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Installed extension
2. Added account with crypto 
3. Verify STX opt-in-modal
4. Send a smart transaction

## **Screenshots/Recordings**



### **Before**



### **After**

![image](https://github.com/user-attachments/assets/6d610e0d-48b8-43e5-9dff-774481966e77)

![image](https://github.com/user-attachments/assets/4a8f5c17-4e2e-402b-84ca-61c95ed08948)

![image](https://github.com/user-attachments/assets/0eae012c-4cfe-4272-94c2-12b7519af1cb)

![image](https://github.com/user-attachments/assets/2cd232e6-d279-4894-b2df-5fcee0a8a0c9)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
